test/indexer/test_indexer_controller.py

```diff
@@ -0,0 +1,151 @@
+        self.assertRaises(AssertionError, self.controller.contribute, event)
+
+    def test_contribution_action_reindex(self):
+        event = [self.generate_sqs_record(action='reindex', prefix='ff')]
+        with mock.patch('azul.azulclient.AzulClient.do_remote_reindex') as mock_reindex:
+            mock_reindex.return_value = True
+            self.controller.contribute(event)
+            self.assertEqual(mock_reindex.call_count, 1)
+
+    def test_contribute_and_aggregate(self):
+        """
+        Index two bundles that make contributions to the same project.
+        Inspect that the contributions match the tallies that are returned to
+        SQS. During aggregation only the project entity is deferred due to
+        multiple contributions.
+        """
+        self._create_mock_queues()
+        event = []
+        bundles = []
+        contributions_per_bundle: Sequence[Sequence[Contribution]] = []
+        bundle_fqids = [
+            BundleFQID('56a338fe-7554-4b5d-96a2-7df127a7640b', '2018-03-28T151023.074974Z'),
+            BundleFQID('b2216048-7eaa-45f4-8077-5a3fb4204953', '2018-03-29T104041.822717Z')
+        ]
+        for bundle_fqid in bundle_fqids:
+            notification = self.client.synthesize_notification(self.catalog, bundle_fqid)
+            event.append(self.generate_sqs_record(action='add',
+                                                  catalog=self.catalog,
+                                                  notification=notification))
+            bundle = self._load_canned_bundle(bundle_fqid)
+            bundles.append(bundle)
+            contributions_per_bundle.append(self.index_service.transform(self.catalog, bundle=bundle, delete=False))
+        repository_patch = mock.patch('azul.indexer.index_controller.IndexController.repository_plugin')
+        service_patch = mock.patch('azul.indexer.index_controller.IndexController.index_service',
+                                   new_callable=mock.PropertyMock)
+        with repository_patch as mock_repository, service_patch as mock_index_service:
+            mock_index_service.return_value = ForcedRefreshIndexService()
+            mock_repository.return_value = mock.MagicMock()
+            mock_repository.return_value.fetch_bundle.side_effect = bundles
+            with self.subTest('Test contributions from 2 bundles with matching project_ids'):
+                self.controller.contribute(event)
+                tallies = [
+                    json.loads(m.body)
+                    for m in self.queue_manager.read_messages(self.controller._tallies_queue())
+                ]
+                entities_from_tallies = {(t['entity_id'], t['entity_type']) for t in tallies}
+                entities_from_contributions = {
+                    (contributions.entity.entity_id, contributions.entity.entity_type)
+                    for cl in contributions_per_bundle for contributions in cl
```

Comment:
I don't know what `cl` stands for. Also for `contributions`, isn't it just a single contribution at that point?

If I'm correct, then I think
```suggestion
                    for cs in contributions_per_bundle for c in cs
```
makes more sense.